README.md

```diff
@@ -128,6 +128,42 @@ SolidusStripe.configure do |config|
 end
 ```
 
+## Implementation
+
+### Payment state-machine vs. PaymentIntent statuses
+
+When compared to the Payment state machine, Stripe payment intents have different set of states and transitions.
+The most important difference is that on Stripe a failure is not a final state, rather just a way to start over.
+
+In order to map these concepts SolidusStripe will match states in a slightly unexpected way, as shown below.
+
+Reference: https://stripe.com/docs/payments/intents?intent=payment
+
+![image](https://user-images.githubusercontent.com/1051/217322027-f49081f5-0795-49f4-994e-285a9de5347c.png)
+
+### ⚠️ Warning: Authorization happens before the order is completed
+
+This setup implies that after the payment information are submitted to Stripe the payment is authorized, although
+the order is not yet completed. This can become an issue if a customers abandons the checkout at the `confirm`
+step and no action is taken to free up the money on the backend.
+
+In order to mitigate this issue we suggest to adapt the frontend merging the confirm and payment steps. Embedding
+the agreement to the terms of service and an order summary in the payment step and apply this patch:
+
+```patch
+--- a/templates/app/controllers/checkouts_controller.rb
++++ b/templates/app/controllers/checkouts_controller.rb
+@@ -48,6 +48,8 @@ def redirect_on_failure
+   end
+
+   def transition_forward
++    @order.next if @order.has_checkout_step?("payment") && @order.payment?
++
+     if @order.can_complete?
+       @order.complete
+     else
+```
+
```

Comment:
Considering that there might be some unexpected problems when merging both checkout steps, what do you think if we propose creating a background job as another option?